src/Tomcat/TomcatController.ts

```diff
@@ -157,7 +157,7 @@ export class TomcatController {
             }
             server.needRestart = restart;
             await Utility.trackTelemetryStep(operationId, restart ? 'restart' : 'stop', () =>
-                Utility.executeCMD(this._outputChannel, server.getName(), Utility.getJavaExecutable(), { shell: true }, ...server.jvmOptions.concat('stop')));
+                Utility.executeCMD(this._outputChannel, server.getName(), Utility.getJavaExecutable(), Utility.getJavaEnvironment({ shell: true }), ...server.jvmOptions.concat('stop')));
```

Comment:
No need to pass spawnOptions when you calculate custom environment variables. Something like `Utility.executeCMD(.... {shell: true, env: Utility.getCustomEnv()}, ....)` would be clearer.